pkg/arvo/sys/zuse.hoon

```diff
@@ -4291,15 +4342,76 @@
     ::                                                  ::  ++stri:de-json:html
     ++  stri                                            ::  string
       (cook crip (ifix [doq doq] (star jcha)))
-    ::                                                  ::  ++tops:de-json:html
-    ++  tops                                            ::  strict value
-      ;~(pose abox obox)
     ::                                                  ::  ++spac:de-json:html
     ++  spac                                            ::  whitespace
       (star (mask [`@`9 `@`10 `@`13 ' ' ~]))
-    ::                                                  ::  ++twel:de-json:html
-    ++  twel                                            ::  tape weld
-      |=([a=tape b=tape] (weld a b))
+    ::                                                  ::  ++tops:de-json:html
+    ++  tops                                            ::  strict value
+      ;~(pose abox obox)
+    ::                                                  ::  ++unix:de-json:html
+    ++  unic                                            ::  UTF-16 unicode escape
+      (bass 16 (stun [4 4] hit))
+    ::                                                  ::  ++utfe:de-json:html
+    ++  utfe                                            ::  UTF-8 sequence
+      ;~  less  doq  bas
+        =*  qua
+          %+  cook
+          |=  [a=@ b=@ c=@ d=@]
+            :((cury cat 3) a b c d)
+          ;~  pose
+            ;~  plug
+              (shim 241 243)
+              (shim 128 191)
+              (shim 128 191)
+              (shim 128 191)
+            ==
+            ;~  plug
+              (just '\F0')
+              (shim 144 191)
+              (shim 128 191)
+              (shim 128 191)
+            ==
+            ;~  plug
+              (just '\F4')
+              (shim 128 143)
+              (shim 128 191)
+              (shim 128 191)
+            ==
+          ==
+        =*  tre
+          %+  cook
+          |=  [a=@ b=@ c=@]
+            :((cury cat 3) a b c)
+          ;~  pose
+            ;~  plug
+              ;~  pose
+                (shim 225 236)
+                (shim 238 239)
+              ==
+              (shim 128 191)
+              (shim 128 191)
+            ==
+            ;~  plug
+              (just '\E0')
+              (shim 160 191)
+              (shim 128 191)
+            ==
+            ;~  plug
+              (just '\ED')
+              (shim 128 159)
+              (shim 128 191)
+            ==
+          ==
+        =*  dos
+          %+  cook
+          |=  [a=@ b=@]
+            (cat 3 a b)
+          ;~  plug
+            (shim 194 223)
+            (shim 128 191)
+          ==
+        ;~(pose qua tre dos)
+      ==
```

Comment:
If this goes in at all, I'd put it in https://github.com/urbit/urbit/blob/bbd0441/pkg/arvo/sys/hoon.hoon#L4277 and call out to that from here. (Or rather, from whatever is calling here repeatedly, have it collect `;~(less doq bas prn)` and then `sand` the resulting `[%s @t]`)

Reply:
1. @joemfb No, the UTF8 and UTF16 parsing are independent. Originally, the Hoon version of `+de-json:html` was parsing whatever (which is technically acceptable, as the JSON standard is ambiguous about what to do for malformed UTF8/UTF16). However, when I was adding the jet for `de-json`, the C parser from which I chose to copy has pretty involved UTF8/UTF16 parsing with which I didn't want to mess. This was for two reasons: I didn't want to deal with it as it would have involved a lot of changes and (as discussed below) I was wondering if there was a way that we might incorporate the parser as a sub-module so that we don't have to have our own copy of it in the jet code. Therefore, it seemed more simple to make the Hoon version of `de-json` only accept correct UTF8/UTF16.
2. @joemfb Why is the call to `sane` in `ruth` commented out?
3. What @ohAitch suggests seems like the correct decision to me: move an optimized version of the UTF8 validation into `sane` (or wherever is appropriate, then call it from `sane`), parse any text, and verify it with `sand`

Reply:
There are two questions at issue here:

- since `cord` is already (supposed to be) utf-8, is it worth the optimization of dropping the utf-8 parser/validator from the json parser?
- if we are going to parse/validate utf-8 in the json parser, how should it be implemented and where should it live?

In theory, we should not have to (re-)parse utf-8 at this layer (ie, keep the old behavior of `+jcha`: `;~(pose ;~(less doq bas prn) esca)`). In practice, not doing so requires us to fork this json parser (and own it going forward). It would be much nicer to just include it as a dependency (or vendor it) and not make deep changes. @ohAitch, do you have thoughts about what's righteous to specify in hoon/nock at this layer?

If we decide to parse/validate utf-8, it should be enough to call `(sane %t)` from `+jcha`, changing it to something like:

```
;~(pose (sear |=(a=@t ?.(((sane %t) ~ `a)) (cook crip ;~(less doq bas prn)) (cook crip esca))
```

(and removing `(cook crip ...)` from `+stri`).

As for `+ruth`, I don't know if that validation was ever wired up. To do so, we need to jet (and fill out) `+sane`, then track down and fix any aura violations we're depending on.